core/src/test/java/jenkins/util/UrlHelperTest.java

```diff
@@ -10,56 +10,140 @@
 
     @Test
     @Issue("JENKINS-31661")
-    public void isValid() {
+    public void regularCases() {
```

Comment:
It's up to you on style, but this type of test may benefit from using a parameterized test. It also helps make explicit that this test is actually more like 30 tests rolled into one. :)

Reply:
If we were in Groovy / Spock, I would like to use some parameterized test but in pure Java / Junit4, the amount of code is too big to gain something in terms of readability from my PoV.

Reply:
JUnit 5 is compatible with v4 and v3. The parameterized testing feature there is a lot less verbose and allows for method parameter injection as well. Might be worth looking at sometime.